#!/usr/bin/env python
PACKAGE = "proc_control"

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("T1_X", double_t, 0, "T1: The force in X", 0, -1, 1)
gen.add("T1_Y", double_t, 0, "T1: The force in Y", 0, -1, 1)
gen.add("T1_Z", double_t, 0, "T1: The force in Z", 0, -1, 1)
gen.add("T1_YAW", double_t, 0, "T1: The force in YAW", 0, -1, 1)
gen.add("T1_PITCH", double_t, 0, "T1: The force in PITCH", 0, -1, 1)
gen.add("T1_ROLL", double_t, 0, "T1: The force in ROLL", 0, -1, 1)

gen.add("T2_X", double_t, 0, "T2: The force in X", 0, -1, 1)
gen.add("T2_Y", double_t, 0, "T2: The force in Y", 0, -1, 1)
gen.add("T2_Z", double_t, 0, "T2: The force in Z", 0, -1, 1)
gen.add("T2_YAW", double_t, 0, "T2: The force in YAW", 0, -1, 1)
gen.add("T2_PITCH", double_t, 0, "T2: The force in PITCH", 0, -1, 1)
gen.add("T2_ROLL", double_t, 0, "T2: The force in ROLL", 0, -1, 1)

gen.add("T3_X", double_t, 0, "T3: The force in X", 0, -1, 1)
gen.add("T3_Y", double_t, 0, "T3: The force in Y", 0, -1, 1)
gen.add("T3_Z", double_t, 0, "T3: The force in Z", 0, -1, 1)
gen.add("T3_YAW", double_t, 0, "T3: The force in YAW", 0, -1, 1)
gen.add("T3_PITCH", double_t, 0, "T3: The force in PITCH", 0, -1, 1)
gen.add("T3_ROLL", double_t, 0, "T3: The force in ROLL", 0, -1, 1)

gen.add("T4_X", double_t, 0, "T4: The force in X", 0, -1, 1)
gen.add("T4_Y", double_t, 0, "T4: The force in Y", 0, -1, 1)
gen.add("T4_Z", double_t, 0, "T4: The force in Z", 0, -1, 1)
gen.add("T4_YAW", double_t, 0, "T4: The force in YAW", 0, -1, 1)
gen.add("T4_PITCH", double_t, 0, "T4: The force in PITCH", 0, -1, 1)
gen.add("T4_ROLL", double_t, 0, "T4: The force in ROLL", 0, -1, 1)

gen.add("T5_X", double_t, 0, "T5: The force in X", 0, -1, 1)
gen.add("T5_Y", double_t, 0, "T5: The force in Y", 0, -1, 1)
gen.add("T5_Z", double_t, 0, "T5: The force in Z", 0, -1, 1)
gen.add("T5_YAW", double_t, 0, "T5: The force in YAW", 0, -1, 1)
gen.add("T5_PITCH", double_t, 0, "T5: The force in PITCH", 0, -1, 1)
gen.add("T5_ROLL", double_t, 0, "T5: The force in ROLL", 0, -1, 1)

gen.add("T6_X", double_t, 0, "T6: The force in X", 0, -1, 1)
gen.add("T6_Y", double_t, 0, "T6: The force in Y", 0, -1, 1)
gen.add("T6_Z", double_t, 0, "T6: The force in Z", 0, -1, 1)
gen.add("T6_YAW", double_t, 0, "T6: The force in YAW", 0, -1, 1)
gen.add("T6_PITCH", double_t, 0, "T6: The force in PITCH", 0, -1, 1)
gen.add("T6_ROLL", double_t, 0, "T6: The force in ROLL", 0, -1, 1)

gen.add("T7_X", double_t, 0, "T7: The force in X", 0, -1, 1)
gen.add("T7_Y", double_t, 0, "T7: The force in Y", 0, -1, 1)
gen.add("T7_Z", double_t, 0, "T7: The force in Z", 0, -1, 1)
gen.add("T7_YAW", double_t, 0, "T7: The force in YAW", 0, -1, 1)
gen.add("T7_PITCH", double_t, 0, "T7: The force in PITCH", 0, -1, 1)
gen.add("T7_ROLL", double_t, 0, "T7: The force in ROLL", 0, -1, 1)

gen.add("T8_X", double_t, 0, "T8: The force in X", 0, -1, 1)
gen.add("T8_Y", double_t, 0, "T8: The force in Y", 0, -1, 1)
gen.add("T8_Z", double_t, 0, "T8: The force in Z", 0, -1, 1)
gen.add("T8_YAW", double_t, 0, "T8: The force in YAW", 0, -1, 1)
gen.add("T8_PITCH", double_t, 0, "T8: The force in PITCH", 0, -1, 1)
gen.add("T8_ROLL", double_t, 0, "T8: The force in ROLL", 0, -1, 1)

exit(gen.generate(PACKAGE, "proc_control", "Thruster"))